src/Evolution/Systems/Cce/Actions/SendNextTimeToCce.hpp

```diff
@@ -0,0 +1,152 @@
+ * full step.
+ *
+ * Uses:
+ * - DataBox:
+ *   - `Tags::TimeStepId`
+ *   - `Tags::Next<Tags::TimeStepId>`
+ *   - `Tags::TimeStepper<TimeStepper>`
+ *
+ * \ref DataBoxGroup changes:
+ * - Adds: nothing
+ * - Removes: nothing
+ * - Modifies: nothing
+ */
+template <typename InterpolationTargetTag>
+struct SendNextTimeToCce {
+  using const_global_cache_tags = tmpl::list<::Tags::TimeStepper<TimeStepper>>;
+  template <typename DbTags, typename Metavariables, typename... InboxTags,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static std::tuple<db::DataBox<DbTags>&&> apply(
+      db::DataBox<DbTags>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      Parallel::ConstGlobalCache<Metavariables>& cache,
+      const ArrayIndex& array_index, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) noexcept {
+    const auto& block_logical_coords =
+        get<intrp::Vars::PointInfoTag<InterpolationTargetTag,
+                                      Metavariables::volume_dim>>(
+            db::get<intrp::Tags::InterpPointInfo<Metavariables>>(box));
+    const std::vector<ElementId<Metavariables::volume_dim>> element_ids{
+        {array_index}};
+    typename intrp::Vars::PointInfoTag<InterpolationTargetTag,
+                                       Metavariables::volume_dim>::type
+        first_valid_block_logical_coords;
+    for (const auto& coordinate : block_logical_coords) {
+      if (static_cast<bool>(coordinate)) {
+        first_valid_block_logical_coords.push_back(coordinate);
+        break;
+      }
+    }
+    if (first_valid_block_logical_coords.empty() ) {
+      Parallel::printf(
+          "Warning: No valid block logical coordinates found, please ensure "
+          "that at least one point for the interpolation is within the domain "
+          "to use `SendNextTimeToCce`. No 'next' time is sent, but execution "
+          "will continue.\n");
+    }
+    const auto element_coord_holders = element_logical_coordinates(
+        element_ids, first_valid_block_logical_coords);
+
+    if (LIKELY(element_coord_holders.count(element_ids[0]) == 0)) {
+      return std::forward_as_tuple(std::move(box));
+    }
+    if (db::get<::Tags::TimeStepId>(box).substep() == 0) {
+      // find the next step time via the timestepper
+      auto next_step_id = db::get<::Tags::Next<::Tags::TimeStepId>>(box);
+      while (next_step_id.substep() != 0) {
+        next_step_id =
+            db::get<::Tags::TimeStepper<TimeStepper>>(box).next_time_id(
+                next_step_id, db::get<::Tags::TimeStep>(box));
+      }
+      auto& receiver_proxy = Parallel::get_parallel_component<
+          typename Metavariables::cce_boundary_component>(cache);
+      Parallel::simple_action<Actions::ReceiveNextElementTime>(
+          receiver_proxy, db::get<::Tags::TimeStepId>(box),
+          std::move(next_step_id));
+      return std::forward_as_tuple(std::move(box));
```

Comment:
Is this line needed? Would omitting it not go to the same statement on line 100?